### ISSUE TITLE
Old PT files' layout issues

### ISSUE THREAD

**Issue by theczar** [powertab/powertabeditor#164]
Quite often when opening an older PT file the layout gets arranged wrong way. If the original file had 4 measures per line, the "new" layout will  somehow force it to 3 measures per line, which will screw up the whole tab structure.

**Comment by cameronwhite** [comment]
What you're seeing is probably a result of merging the bass score with the guitar score, but the importer tries to preserve the original layout as much as possible. If you have some example files, it would be useful to see cases where improvements can be made.

**Comment by rekado** [comment]
I have a couple of old PT files where the bass score appears to be shifted after a couple of repeats.  

In the old PT editor it was possible to assign different repeat endings to the bass score and the guitar score.  I'm not sure if a lack of support for this feature is what causes scores to fall apart after opening them in PTE2.

I can try to find a simple example among my stash of tabs for PTE1.7.

**Comment by cameronwhite** [comment]
Yeah, I suspect the score merger doesn't properly handle cases where the scores are both in a repeated section but have different repeat endings.

**Comment by greatzot** [comment]
I ran into a similar problem today with an old score. The intro of a song was notated for guitar as two sections with a four-measure repeating structure, and two different repeat endings, so a total of 10 measures visually, or 16 in playback terms. In addition, there was an additional 2:4 measure following all of that to sustain the last note and countoff into the song body. So for the guitar score, the first staff/system/line had 5 measures, and the second 6, for 11 measures visually, or 17 in playback terms.

But in the bass score, the bass plays nothing for the first 8 measures, so that was notated as an 8-bar rest, and instead of bothering to write the next bit with repeat endings, I had simply written out all 8 measures plus a 2:4 measure. 

When I imported to Power Tab 2.0, I got 17 measures, including that 2:4 measure at the end, but with the 8th and 16th showing a repeat bar line, and measure 8 showing a repeat ending labeled 1. No repeat barline was shown at the beginning of either section, and no other repeat ending marks. Through all of that, it went with a 4-measure-per-system layout, but then after that switched to 3, and later varied from 3 to 2 to 4 but never more than 4.

After reading this thread, I tried making the guitar and bass score intros consistent with each other, but again, when importing, it doesn't use the 5- or 6-measure per line layout I had for the first two systems, even after I made the bass and guitar consistent there. It just maxes out at 4 and then pushes the rest to the next line, giving a very different layout overall from the original score.

Just to be sure, I even went so far as to remove the bass score entirely, to see if anything changed. Nope--still got a max of 4 measures per line.

So it looks to me like there's more going on than just whether the bass and guitar scores match--it also seems to limit the score to 4 measures per line.

One suggestion I have that might help in some of these scenarios is, since the original Power Tab effectively treated the guitar and bass as two separate scores in terms of how they were written, and just played back whatever you wrote together like two different instruments in MIDI, it might be better to provide the user an option to either merge both scores into one, as it does now, or open the guitar score in one file and the bass score in another file. (Which would also be a step ahead of Guitar Pro, as that forces you to choose between either merging or only opening one score, rather than opening both scores in two new, separate files right off the bat.) That way you can choose what's more important to you--the visual layout previously created for each, or keeping it all together in one file.

**Comment by cameronwhite** [comment]
Yeah, having an option to avoid merging the scores is probably a good idea. It's not too hard to skip the merging internally (https://github.com/powertab/powertabeditor/blob/8f9fafbd9fdf1bfde85916ca711acb75d47fa334/source/formats/powertab_old/powertaboldimporter.cpp#L80) so it's mostly a matter of adding the UI around it and passing down some configuration options for which score to load
